fix(core): set `npm_package_json` environment variable when running scripts (#4546)
yarnpkg/berry: packages/gatsby/content/advanced/lifecycle-scripts.md

- `$npm_package_version` is its version.

> - `$npm_package_json` is the path of the project's `package.json` on the filesystem.

Reviewer: ```suggestion
- `$npm_package_json` is the absolute path to its `package.json`.
```
